NarcissisticNumbers solution and test case.

zulqer-nain opened the pull request:
Solution to find Narcissistic numbers in range .
Solution to find if provided number is Narcissistic or not .
Test cases to check whether provided number is Narcissistic or not .
Test cases to find Narcissistic numbers in provided range .
fixed #11

zulqer-nain committed:
NarcissisticNumbers solution and test case.

Solution to find Narcisstic numbers in range .
Solution to find if provided number is Narcisstic or not .
Test cases to check whether provided number is Narcisstic or not .
Test cases to find Narcisstic numbers in provided range .

nirajunify commented on src/zulqer/narcissisticNumbers/__tests__/index.test.ts:
```diff
@@ -0,0 +1,29 @@
+import { getNarcissisticNumbersInRange, isNarcissisticNumber } from '..';
+
+describe('find if provided number is Narcissistic using isNarcissisticNumber function ', function () {
```
Since, you are already using anonymous function, wouldn't it be nice if you replace that with arrow function?

zulqer-nain replied:
yes .

nirajunify commented on src/zulqer/narcissisticNumbers/__tests__/index.test.ts:
```diff
@@ -0,0 +1,29 @@
+
+  it('10 is non NarcissisticNumber ', function () {
+    expect(isNarcissisticNumber(10)).toBe(false);
+  });
```
You should add a negative number test

zulqer-nain replied:
alright

nirajunify commented on src/zulqer/narcissisticNumbers/__tests__/index.test.ts:
```diff
@@ -0,0 +1,29 @@
+describe('find narcissistic numbers in provided range using getNarcissisticNumbersInRange function  ', function () {
+  it('narcissistic numbers between 10 to 100', function () {
+    expect(getNarcissisticNumbersInRange(10, 100)).toEqual([]);
+  });
```
Add space

zulqer-nain replied:
between lines ?

nirajunify replied:
yes

zulqer-nain replied:
unnecessary lines will be removed by prettier

nirajunify commented on src/zulqer/narcissisticNumbers/index.ts:
```diff
@@ -0,0 +1,26 @@
+  for (let i = 0; i < length; i++) {
+    sum += Math.pow(parseInt(arrayOfCurrentNumber[i]), length);
+  }
+  if (sum === num) {
```
You can just return this comparison

zulqer-nain committed:
Resolved Comments on PR

Replaced annonymus function with arrow function .
Added negative number test case.
Added Negative Range test case.
Improvised isNarcissisticNumber function .

zulqer-nain reviewed (commented):
Resolved comments.

nirajunify reviewed (dismissed):
I still believe you can improve your test cases. However, that's continuous process, so I am approving this PR with the belief that you would keep on reading other articles on how to write better test cases.

Note: If you find some good reference. Please let the team know

zulqer-nain commented:
Sure.

krajasekhar-unify commented on src/zulqer/narcissisticNumbers/index.ts:
```diff
@@ -0,0 +1,22 @@
+  return sum === num;
+}
+
+function getNarcissisticNumbersInRange(
```
Lets please stick to writing ES6 arrow functions to define a function.
```
const getNarcissisticNumbersInRange = () => {}
```

krajasekhar-unify commented on src/zulqer/narcissisticNumbers/index.ts:
```diff
@@ -0,0 +1,22 @@
+function isNarcissisticNumber(num: number): boolean {
+  const arrayOfCurrentNumber = String(num).split('');
+  const length = arrayOfCurrentNumber.length;
```
Using `length` property from the array every time instead of storing it in a variable is no more a bad way. There is no such performance issue in JS as length is just a property of an array and it won't execute any code to find the length.

krajasekhar-unify replied:
May be we can still do this when using it too many times just to reduce the variable width.

zulqer-nain committed:
Resolved comments

Changed function to arrow function.
Accessed length property directly insted of storing in variable

zulqer-nain reviewed (commented):
Added requested changes @krajasekhar-unify

nirajunify reviewed (approved):
Approving
